### PR TITLE
[Chore] Cleanup parse_data_from_io legacy/segwit parsing

### PR DESCRIPTION
**Pull request by zquestz**
This just cleans up the `parse_data_from_io` method and makes it more readable.

I am new to this gem, so let me know if I messed anything up. ;)

### CODE REVIEW

**Comment by etscrivner** [comment]
LGTM

**Comment by zquestz** [comment]
Ok all tests are now passing. Fixed the issue.

Docs:
https://bitcoin.org/en/developer-reference#raw-transaction-format
https://github.com/bitcoin/bips/blob/master/bip-0144.mediawiki

File: lib/bitcoin/protocol/tx.rb [hunk 1]
```diff
@@ -81,23 +81,16 @@ def parse_data_from_io(data)
...
-          # marker must be zero and flag must be non-zero to be valid segwit
-          if @marker == 0 && @flag != 0
-            in_size = Protocol.unpack_var_int_from_io(buf)
-            witness = true
-          else
-            # Undo the last read in case this isn't a segwit transaction
-            buf.seek(buf.pos - 1)
-          end
-        end
+        @marker = buf.read(1).unpack('c').first
```
Comment (jimpo) [inline lib/bitcoin/protocol/tx.rb]: `[@marker, @flag] = buf.read(2).unpack('cc')`

**Comment by lian** [comment]
cherry-picked into master

**Pull request closed without merging**